### PR TITLE
PR feedback from ownership cleanliness

### PR DESCRIPTION
**Pull request by czue**
also start actually running tests on the new sync code

@snopoke cc @NoahCarnahan

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
+import functools
 import uuid
 from couchdbkit import ResourceNotFound
 from django.test.utils import override_settings
 from django.test import TestCase
 import os
 from casexml.apps.phone.exceptions import MissingSyncLog, RestoreException
+from corehq.apps.tzmigration.test_utils import run_with_multiple_settings, RunWithMultipleSettings, \
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: F401 'run_with_multiple_settings' imported but unused

**Comment by czue** [comment]
wait for tests now

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py [hunk 2]
```diff
@@ -31,6 +33,67 @@
...
+
+class RunConfig(object):
+
+    def __init__(self, settings, pre_run=None, post_run=None):
+        self.settings = settings
+        self.pre_run = pre_run or NOOP
+        self.post_run = post_run or NOOP
+
+
+class RunWithMultipleConfigs(object):
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: stole a lot of this from `RunWithMultipleSettings` but needed to extend it a bit. wasn't sure it was helpful to push this into there so just copied into a new class. cc @dannyroberts

Comment (dannyroberts) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: I'm ok with that :)

**Pull request merged**